pkg/webhook/board_update.go

```diff
@@ -60,16 +78,91 @@ func (h *boardUpdate) handleIssuesEvent(event *github.IssuesEvent, gh *github.Cl
 	logger.Debug("Issue Action: " + *event.Action)
 
 	eventKey := messageType + "_" + *event.Action
+
+	// post processing (from previous event cycle)
+	// takes precedence, the event will no be processed further
+	if "issues_closed" == eventKey {
+		_, issue_scheduled := postProcessing[number]
+
+		if issue_scheduled {
+			delete(postProcessing, number)
+			go postProcess(event, gh, config, *doneColumn, logger)
+			return nil
+		}
+	} else if "issues_reopened" == eventKey && event.GetIssue().GetLocked() {
+		// move
+		err := moveIssueOnBoard(config, number, *doneColumn, logger)
+
+		// update progress/* label
+		changeProgressLabel(gh, event.Repo, *event.Issue, doneColumn.name)
+
+		if err != nil {
+			logger.Error("Post processing failed: Cannot move issue")
+		}
+	}
+
+	// cleanup post processing markers, but skip actions
+	if event.GetIssue().GetLocked() {
+		gh.Issues.Unlock(context.Background(), event.Repo.Owner.GetLogin(), event.Repo.GetName(), *event.Issue.Number)
+		logger.Debug("Unlock and ignore event for: " + number)
+		return nil
+	}
+
+	// regular processing
 	col, ok := stateMapping[eventKey]
 	if ok {
-		return moveIssueOnBoard(config, number, col, logger)
+		err := moveIssueOnBoard(config, number, col, logger)
+
+		if nil == err {
+			// update progress/* label
+
+			changeProgressLabel(gh, event.Repo, *event.Issue, col.name)
+		}
+
+		return err
 	} else {
-		logger.Debug("Ignore unmapped event: " + eventKey)
+		logger.Debug("Ignore unmapped Issue event: " + eventKey)
 	}
 
 	return nil
 }
 
+func changeProgressLabel(gh *github.Client, repo *github.Repository, issue github.Issue, newLabel string) {
+
+	labels := []string{"progress/" + newLabel}
```

Comment:
If it's not too much of an effort, it would be awesome if we could make this label-prefix configurable like ("progressLabelGroup" or maybe even more comfortable like w "progressLabelPattern" with a default of "progress/%s" and then filling in the column name).

Also, if you don't mind it would be cool if we could lower case the column names when used in labels as we currently have this for all other labels (except "Epic" which has a ZenHub meaning).

Reply:
lower case can be handled in the config. It's matter of using the proper case in `column.Name`. I'll add the prefix option.